scenario/state.py

```diff
@@ -521,6 +698,26 @@ class Container:
 
     exec_mock: _ExecMock = dataclasses.field(default_factory=dict)
 
+    # NOTE: This can be replaced with dataclasses.KW_ONLY in Python 3.10+
+    def __init__(
+        self,
+        name: str,
+        *,
+        can_connect: bool = False,
+        _base_plan: Optional[dict] = None,
+        layers: Optional[Dict[str, pebble.Layer]] = None,
+        service_status: Optional[Dict[str, pebble.ServiceStatus]] = None,
```

Comment:
Side point, but should this be called `service_statuses`? We usually use plural for lists/dicts/sets of things. (But if so, let's do in another PR to avoid muddying the waters.)

Reply:
Agreed. I similarly have two other plural name changes in the [components are sets PR](https://github.com/canonical/ops-scenario/pull/134).

I think this could go into the [exec_mocks->execs PR](https://github.com/canonical/ops-scenario/pull/124) without muddying that one up.